### DISCUSSION TITLE
LFS to Metered Billing FAQ

### DISCUSSION THREAD

**Discussion by tuves** [community/community#61362] [Enterprise]
### What is happening?
Git LFS is moving from a pre-paid, quota-based billing model (data packs) to a post-paid, usage-based billing model (metered billing). Also, GitHub is raising the free, included amount of Git LFS resources to 10 GB storage and bandwidth for Free and Pro accounts and 250 GB storage and bandwidth for Team and Enterprise accounts.

We are shifting to a universal bill date for all products. Billing dates will align on the first day of each month. As an existing GitHub customer, your balance will be prorated once to align with the new universal bill date.

### Why is Git LFS moving to metered billing?
To align with our other product lines that leverage bandwidth and storage, as well as to allow LFS to be Microsoft Azure Commit to Consume (MACC) eligible. 

### What will happen to data packs?
GitHub will discontinue selling data packs.

### When is Git LFS moving to Metered Billing?
On August 31, 2023, GitHub alerted you of our plan to change the billing method for Git Large File Storage (Git LFS) from customer purchased Data Packs to metered billing. Git LFS customers will also be onboarded to the new billing platform.

> [!IMPORTANT]
> **In order to provide the smoothest transition, we will no longer shift the billing method on the original date of October 1, 2023. As the new method and timeline are confirmed, we will provide updates.**

### What is Git LFS?
Large File Storage 

### What is changing with the Git LFS Entitlements?
GitHub is increasing the amount of free storage and bandwidth for Git LFS. GitHub customers with Team and Enterprise plans will receive 250 GB of storage and 250 GB of bandwidth per month. (Yes, that is a 250x increase.) GitHub Free and Pro users will receive 10 GB of storage and 10 GB of bandwidth per month (a 10x increase).

### Will customers receive a refund for unused Data Packs? 
No, however, customers will receive credit for the unused portion of their data packs. We will not be issuing refunds for unused data packs. 

### How can customers check their Git LFS usage?
[Link to instructions](https://docs.github.com/en/billing/managing-billing-for-git-large-file-storage/viewing-your-git-large-file-storage-usage) to view your Git LFS use.

### Will customers have to pay for previously used and unbilled Git LFS use?
No, customers will only receive a bill for their monthly use starting in the month when LFS metered billing takes effect.

### How can customers control their costs for Git LFS?
Like other GitHub features, Git LFS will have a budget you can set to control costs. The budget defaults to zero for customers with no payment method on file. For all other customers, the budget is unlimited. You can take these steps to help manage your Git LFS bill. 

- [Review your LFS usage](https://docs.github.com/en/billing/managing-billing-for-git-large-file-storage/viewing-your-git-large-file-storage-usage)
- Delete Relevant Repositories 
- Remove LFS Objects 
- Set an LFS Budget 

### How much is Git LFS going to cost after the switch?
Each GB of data downloaded will be charged at $0.0875 USD. For storage, pricing will match GitHub Codespaces at $0.07 per GB. Storage is billed by calculating an hourly usage rate. So, for example, assume you used 1 GB above the free entitlement for the first 15 days of April, then used 2 GB starting from the 16th. We would calculate your usage as:

1. 1 GB × 15 days × 24 hours per day = 360 GB-hours
2. 2 GB × 15 days × 24 hours per day = 720 GB-hours
3. 360 GB-hours + 720 GB-hours = 1080 GB-hours
4. 1080 GB-hours / 720 hours in the month = 1.5 GB-months

Therefore, we would bill for 1.5 GB storage for the month of April.

### How will my bill change?
Coming soon, visit the [GitHub Pricing Calculator](https://github.com/pricing/calculator), where you can calculate your LFS usage costs.

### I see some places that use GB and others that use GB. Why?
GitHub has always internally used GB (1024^3 bytes) when measuring LFS. Previously, we used the abbreviation “GB” when talking about this externally but are taking this opportunity to also be more precise with our terminology. “GB” properly refers to a power of 10 (1000^3 bytes). Nothing is changing in how we actually measure or account for bytes stored and transmitted, only in how we publicly refer to these quantities.

> ### ℹ️ Community Manager Update
> This discussion was updated on Sept. 27, 2023:
> - Removed date from `What is happening` section
> - Updated `When is Git LFS moving to metered billing` section
> - Removed date from `What will happen to data packs?` section
>
> This discussion was updated on May 22, 2024:
> - Removed additional mention of October 1 – the old date when billing would commence.
> - Highlighted that October 1 isn't the date anymore using the "important" Markdown alert.
> - Added "Pro" to the account type that new free entitlements apply to.
> - Added "coming soon" to the part about the pricing calculator because metered LFS pricing calculation isn't there quite yet.

**Comment by ericklove8e5** [comment]
****

**Comment by ericklove8e5** [comment]
https://github.com/ericklove8e5

**Comment by ericklove8e5** [comment]
****

**Comment by ericklove8e5** [comment]
https://github.com/ericklove8e5@[](https://github.com/ericklove8e5)

**Comment by ericklove8e5** [comment]
- [ ] - - [ ] @masachs @zackfern

**Comment by ericklove8e5** [comment]
Inc**[github****](https://github.com/ericklove8e5)**

**Comment by zxcvb51226** [comment]
Sent from my iPadOn Jul 22, 2023, at 6:23 PM, ericklove8e5 ***@***.***> wrote:﻿

**Reply by ericklove8e5** [reply]
yeah hello

**Comment by benhatton-unsw-edu-au** [comment]
Hi,
Can I just flag that under an enterprise account, we desperately need a way to enable and cap LFS spend on a per-organisation basis; better if the billing can be set independently for each organisation to avoid internal accounting overheads.
Cheers, Ben

**Comment by sreebaily** [comment]
@benhatton-unsw-edu-au - We are addressing this issue as apart of Billing vNext. We are currently working on it and should be available in fall this year. Happy to jump on a call to discuss further. Thank you.

**Reply by benhatton-unsw-edu-au** [reply]
Thanks Sree, that's great news.  If you need customer insight into requirements then I am at your call.

**Reply by jobywalker** [reply]
It is now November, did a roll out of this improved billing happen? Or has it been delayed?

**Comment by RyStanley89** [comment]
Hello! Sounds like a great change. Do we know when we can communicate this with customers? I had a ticket today funnily enough on this exact topic. I told the customer about the old data packs pricing model, but I'm wondering if I can share any of this?

**Reply by vtbassmatt** [reply]
Customer communications went out this week by email to any customer who will still receive a bill after the changes. (Many customers' usage will be covered by the newly-enlarged free tier 🎉)

**Comment by WoodenMaiden** [comment]
Hello! :wave: 
Thanks for the details. Does the billing apply yearly or monthly?

**Comment by CyberFlameGO** [comment]
Will the spending caps still exist?

**Reply by sreebaily** [reply]
Yes, they will be call as 'Budgets' going forward.

**Comment by const-aurecon** [comment]
Do archived repositories count towards the quota?

**Reply by vtbassmatt** [reply]
Yes. This will be unchanged from today's data pack behavior.

**Comment by qwerfd2-wustl** [comment]
Please allow LFS usage to be optionally metered to the ones who are using the quota, rather than the repo owner. There are public archived repos out there that are non-functional because of the metering rules. You guys want to get paid - that's the goal. Does it really matter who is going to pay you?

**Comment by mehulrajdev** [comment]
Hello! 👋
Thanks for the details. Does the billing apply yearly or monthly?

**Comment by randolphcyg** [comment]
[about-billing-for-git-large-file-storage](https://docs.github.com/zh/billing/managing-billing-for-git-large-file-storage/about-billing-for-git-large-file-storage)

The Chinese version of the document translates October to September

**Reply by vtbassmatt** [reply]
Thank you! That may be a delay in re-translating, as the date moved from September to October recently.

**Comment by PGSCOM** [comment]
Will Free Teams also have 10GB?

**Comment by rfriedel** [comment]
At the moment we can only use a single Azure subscription for all metered features (billing cap) which bares the risk that the "overusage" of a single feature can disable our ability to use any other feature.  It would be rather useful having the ability to define individual caps by feature to avoid that heavy LFS usage could bring Actions usage down.

Ideally, we'd have the ability to assign more than 1 subscription to a feature with a cap per subscription.

**Comment by xieyuheng** [comment]
Please bring back the old feed!

**Comment by muhammadAhmed2223** [comment]
Will the LFS Storage be billed once, or for every month the repo and LFS files are being maintained by GitHub?

**Reply by vtbassmatt** [reply]
LFS storage is an ongoing cost (both today and after the switch to metered billing).

**Comment by GabrielArcasa** [comment]
Can you examplify the Cost with values? Like if i use 10 GiB, how much will Cost per month? i see the calculus that you made, but i don't really understand.

**Comment by MileyHollenberg** [comment]
Hi there, I just randomly found out about these changes, don't mind them as they look fair to me but I was never contacted about these changes on my organization account and they are set to go live in about 2 weeks already. I will fall back into the free tier (currently using about 1.6Gb of LFS data so I will be well within the 10Gb free limit) but I'm pretty sure I should have gotten an E-Mail about it at least

**Comment by ResilientSpring** [comment]
> What will happen to data packs?
GitHub will discontinue selling data packs on October 1, 2023.

> What is changing with the Git LFS Entitlements?
GitHub Free and Pro users will receive 10GiB of storage and 10GiB of bandwidth per month (a 10x increase).

My Git LFS Data Storage is 1.62 GB of 50 GB, but "Setting > Billing and plans > Plans and usage > Billing Summary" still shows:

Your next payment

$5.00 Due by October 02, 2023

Your recurring charges & discounts

GitHub LFS                  $5.00/month                      On Oct 02, 2023

**Comment by muhammadAhmed2223** [comment]
"When you commit and push a change to a file tracked with Git LFS, a new version of the entire file is pushed and the total file size is counted against the repository owner's storage limit."
Hi. Currently this rule seems quite unfair because I am being charged for storage of an entirely new file when I only changed the current one. This will increase the LFS Storage and hence I will be billed more. Will this rule change so that the total LFS storage size doesn't exceed the actual size of the project I have on my disk? Or atleast we should have the option if we either wish to maintain previous file versions or only the latest versions. Thanks

**Comment by BrainSlayer** [comment]
i have a bigger problem. i added the payment method, billing details. still getting the lfs quota error. nothing is charged and of course i added also a amount of money. system is fucked up and github is not usable right now

**Comment by ettaboyle** [comment]
### ℹ️ Community Manager Update
This discussion was updated on Sept. 27, 2023. 
- Removed date from `What is happening` section
- Updated `When is Git FS moving to metered billing` section
- Removed date from `What will happen to data packs?` section

**Reply by BrainSlayer** [reply]
doesnt matter. doesnt work anyway. consider that i'm maintaining a github.com/mirror account. this changed fucked it up. lfs doesnt work anymore.

**Reply by jsoref** [reply]
> ### Will customers have to pay for previously used and unbilled Git LFS use?
> No, customers will only receive a bill for their monthly use starting in October, so they will receive their 1st bill in October 2023.

This should be updated too.

**Comment by KoolieAid** [comment]
Kind of a bummer that the date was changed. I was planning to migrate a project from GitLab to GitHub

**Comment by V0ldek** [comment]
What is the new date?

**Comment by spianmo** [comment]
As of October 8, the current LFS quota seems to be still 1GB.

**Comment by sparker-dg** [comment]
On the Enterprise Settings --> Billing page, what exactly do those numbers for Actions, Packages, Storage for Actions and Packages, Codespaces Monthly Usage, Copilot monthly usage and GitLFS represent. Is it from start of month to when I run the report? For the last 30 days? For the previous full-month? And how do I get these for months before that so I can record and start to spot trends where we might have to pay for more in the future? 

And, are there any updates on enforcement dates? Financial planning changes like this require large lead times in big companies. 

Thanks!

**Comment by V0ldek** [comment]
Can we at least get an update putting it anywhere in time, @ettaboyle? Is it likely this is going to happen this year? Q12024? Ever?

**Comment by eclipse0922** [comment]
Any update for schedule?

**Comment by Dmitry1987** [comment]
Is it possible to use github repository with an external git LFS server for large files? 😄 💸  if the git repo stores only links to files, could it simply store links to files residing on our aws servers or something like that?

**Comment by Miguel-Sanz** [comment]
Hello there! 👋 I just wanted to express my excitement about the upcoming Billing vNext that addresses the need for enabling and capping LFS spend on a per-organization basis. Kudos to the GitHub team for working on this, and I'm looking forward to the added flexibility it will bring to our enterprise account. If you need any customer insights or feedback on requirements, count me in!

**Comment by andreibacanu-uipath** [comment]
Hello, how is the LFS spending named in a billing report?

**Comment by geymin** [comment]
Sooooooooooooooooooooooooooooooooo...

Is it happening?

**Comment by flo-wolf** [comment]
I am now doubting this transition. Hasn't happened on the announced date, and now it's silent around it..,

**Comment by wdamien** [comment]
This is still listed on Githubs public roadmap for a `Q4 2023 – Oct-Dec` release. Grated it's almost the end of Q4 ... but promising nevertheless.

https://github.com/orgs/github/projects/4247/views/1?pane=issue&itemId=184432

**Reply by RemyMachado** [reply]
`Q3 2023` isn't even close to being finished on the roadmap, yet this issue is planned for `Q4 2023`. It seems unlikely to be released anytime soon 😞

**Reply by Hegemege** [reply]
It was updated two weeks ago to say `Q2 2024 – Apr-Jun`

I wish this thread was updated as well by the team with the timeline change, which will probably end up being postponed to Q4 2024 based on this track record.

I guess I'll just keep buying data packs to cover bandwidth.

**Comment by Dmitry1987** [comment]
> I am now doubting this transition. Hasn't happened on the announced date, and now it's silent around it..,

recently talked to a support person about LFS related things, he mentioned this billing change will be rolled out one day but they have no ETA

**Comment by oleflaten** [comment]
A month ago I was able to go into the https://github.com/pricing/calculator and calculate a price estimate for a team with a need for a certain LFS size. It said something like a Team has 250GB LFS included, if I remember correctly.
Now the LFS section is gone! I don't see it mentioned there other than a link at the top, that leads to nowhere.

I am a little worried, since we are starting this project now, and would like to have an idea of what the cost would be, how we will pay it, and when it is available. And if LFS is a viable option with github.com going forward.

**Reply by Dmitry1987** [reply]
You don't really need to push LFS content to github if you have such a large size of data, especially because you'll pay not only for storage but also the bandwidth back and forth from github and your CI runners (in case you run self hosted CI). So that's also costly and need to account for (let's say you pull 200 gigs for each branch build, etc' might get a surprise bill 😄 ).   
  
But you can store LFS in any server, Vultr, DigitalOcean, aws/azure/GCP, whatever works for you. Just deploy Nexus or any of the other solutions listed here https://github.com/git-lfs/git-lfs/wiki/Implementations  and point your repo LFS remote to it, instead of github.com. While your main content remains in github, the LFS will be free since it's not involved at all (you self-host).  For Vultr example (not a financial advice 🤣 just rough estimation) you get an instance with whatever 2-4 cpu, for 24-30$/month, and at that point all your LFS expenses end (without surprise bills) as long as you run CI within same region in Vultr in same network with your LFS server.    
  
The content of github.com repo is free up to 100gb hard limit (confirmed by support representative, I tried to figure it our for myself) so you can also go ahead and push whatever you need as long as single files are <100mb (hard limit per file afaik), and the rest goes to local LFS server. With this setup you end up free-for-free 💯  on github repo and pay only your server costs, and if it's LFS on-premises hardware server then you're Gucci for as long as it runs ✔️  
  
To redirect your LFS from github to a local server just modify `lfs.url` config value
 `git config --global lfs.url https://your-lfs-server-local-address-or-ip/lfs-path`   
  
personally I used Nexus in the past and I think it's a good choice (didn't use for LFS but I see it's supported there now https://help.sonatype.com/repomanager3/nexus-repository-administration/formats/git-lfs-repositories  and Nexus overall is a breeze to install and manage, I liked it).

**Comment by mysterymagination** [comment]
Will there be a way to set a Spending Limit on metered LFS bandwidth like you can for Actions & Packages? I'm not concerned with the storage aspect, but the bandwidth is totally out of my control for public repos and could easily hit 100s of GB of traffic as interest in the repo picks up.

I'm hoping to remove LFS from my project entirely, but while deinitializing LFS for my repo is simple enough the [docs](https://docs.github.com/en/repositories/working-with-files/managing-large-files/removing-files-from-git-large-file-storage#git-lfs-objects-in-your-repository) essentially say that to fully clear your LFS objects server-side you need to nuke the repo (not helpful) or contact customer support (pray for me).
EDIT: it took a week or so to hear back from support, but they were able to help me clear LFS objects server-side easily. The process of rewriting history to remove LFS from the repo turned out to be the hard part, and I documented what I did here: https://politecuriosity.wordpress.com/2024/01/17/removing-git-lfs-from-a-github-repo/

**Comment by alexrp** [comment]
Hi! These changes sound fantastic. However, the fact that LFS checkouts in GitHub Actions workflows count towards the LFS bandwidth usage remains a major blocker for me adopting GitHub's LFS in any of my projects. Do you have any plans to change this?

**Comment by orilevari** [comment]
@tuves Is there any update you can provide on when the billing scheme will change for LFS, as you described above? Our team is incurring high costs on LFS bandwidth and will have likely move away from tracking our test collateral with git if the new pricing scheme isn't rolled out soon.

**Reply by wdamien** [reply]
This is finally rolling out. https://github.blog/changelog/2024-06-03-new-enterprise-accounts-have-metered-billing-for-git-lfs/

Its only in beta for new Enterprise accounts right now. Coming soon to other levels:
> Other Enterprise accounts on GitHub.com, and Free, Pro, and Team accounts, will gain access to the enhanced billing platform in the coming months.

**Comment by JuanMandon** [comment]
i dont get it, are they making this change or not? i still only have 1 gb free lfs and can only buy storage packages

**Reply by flo-wolf** [reply]
I ended up moving my repos to Gitlab, since they offer free LFS (up to a point)
